instance/tests/models/test_openedx_theme_mixins.py

```diff
@@ -0,0 +1,138 @@
+    """
+
+    @staticmethod
+    def make_test_application(instance, user):
+        """
+        Creates BetaTestApplication with test data and sample colors.
+        """
+        application = BetaTestApplication.objects.create(
+            subdomain='test',
+            instance_name='That username is mine',
+            public_contact_email='test@example.com',
+            project_description='test',
+            user=user,
+            instance=instance,
+            main_color='#001122',
+            link_color='#003344',
+            header_bg_color='#caaffe',
+            footer_bg_color='#ffff11',
+            logo='opencraft_logo_small.png',
+            favicon='favicon.ico',
+        )
+        return application
+
+    def test_colors_and_images_applied(self):
+        """
+        Creates a beta application with asks for some colors and logo/favicon, and checks that the generated
+        ansible variables match those colors and images.
+        """
+        # Create objects
+        OpenEdXInstanceFactory(name='Integration - test_colors_applied', deploy_simpletheme=True)
+        instance = OpenEdXInstance.objects.get()
+        user = get_user_model().objects.create_user('betatestuser', 'betatest@example.com')
+        self.make_test_application(instance, user)
+        appserver = make_test_appserver(instance)
+
+        # Test the results
+        self.assertTrue(instance.deploy_simpletheme)
+        # We check 2 times: one time just the theme vars, next whether they're in the final list
+        ansible_theme_vars = instance.get_theme_settings()
+        ansible_vars = appserver.configuration_settings
+        for variables in (ansible_theme_vars, ansible_vars):
+            parsed_vars = yaml.load(variables) or {}
+            expected_settings = {
+                'SIMPLETHEME_ENABLE_DEPLOY': True,
+                'SIMPLETHEME_SASS_OVERRIDES': [
+                    {'variable': 'link-color',
+                     'value': '#003344', },
+                    {'variable': 'header-bg',
+                     'value': '#caaffe', },
+                    {'variable': 'footer-bg',
+                     'value': '#ffff11', },
+                    {'variable': 'button-color',
+                     'value': '#001122', },
+                    {'variable': 'action-primary-bg',
+                     'value': '#001122', },
+                    {'variable': 'action-secondary-bg',
+                     'value': '#001122', },
+                ],
+                'EDXAPP_DEFAULT_SITE_THEME': 'simple-theme',
+                # for SIMPLETHEME_STATIC_FILES_URLS, see below
+            }
+            for ansible_var, value in expected_settings.items():
+                self.assertEqual(value, parsed_vars[ansible_var])
+
+            # Testing of files is a bit special because if we use SWIFT, we will see URLs, but in local tests we have
+            # SWIFT disabled so it's ok to see file names. We allow for both.
```

Comment:
Is this comment still correct? It looks like we are always expecting URLs now.

Reply:
The problem is in tests run locally with .env.test, because there's no SWIFT upload there (and probably there shouldn't be? They're supposed to be local). I think that if I accept only URLs, working non-SWIFT will fail. We could mock the SWIFT storage method but it would be code only for tests run in local. I thought of adding an `if`: „if using SWIFT storage, then check this; if not check this other thing“, would that be clearer?

Reply:
I might be missing something, but in [this line](https://github.com/open-craft/opencraft/pull/235/files/af68a8c23ffb9384340c4a4d8c33a9884485ea3f#diff-4d7dfc4152cfcf430d6487dd0f2ff228R115), it looks like we are asserting that `logo['url']` is in fact a URL pointing to OVH storage? Does the test still work locally?

I actually think it's totally fine to require SWIFT to be configured even when developing locally. We currently require OVH access to be able to spawn VMs anyway. Unless you disagree, I think it's fine to assume SWIFT storage is set up even when running in the devstack, and just remove the comment.

Reply:
It doesn't accept only URLs. In the regexp there's a `?` to make the left URL path optional, so it allows both.
So then I'll remove the regexp thing and accept only URLs, ok? I'll add some comment to .env.test

Reply:
Ah, sorry I see... I knew I was missing something.

If I understand correctly, when running locally and you don't have swift set up `url` entries will be paths to local files. Does the simpletheme role task that's using `get_url` work if you pass it paths to local files?

Reply:
No it doesn't work, it crashed when I tried (ansible's `get_url` expects a URL). The correct way to include local files in the theme would be through the other variable `SIMPLETHEME_STATIC_FILES_DIR`

Reply:
Ok, in that case we just assume SWIFT is always enabled in dev environment.